src/libsyntax/unicode/tables.rs

```diff
@@ -0,0 +1,595 @@
+// Copyright 2012-2014 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+// NOTE: The following code was generated by "src/etc/unicode.py", do not edit directly
```

Comment:
> I've extracted the xid unicode methods into https://github.com/erickt/unicode-xid, with hope that I can get it landed into https://github.com/unicode-rs (ping @kwantam / @SimonSapin).

Pong! I'll have a look.

I assume this means there's a planned story for allowing rustc to depend on crates.io functionality in the future?

It occurs to me that if the XID functions are being removed in the future, we might want to mark them deprecated in librustc_unicode now and direct people to unicode-xid.